libsplinter/src/biome/rest_api/actix/profile.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::actix_web::HttpResponse;
+use crate::futures::IntoFuture;
+use crate::protocol;
+use crate::rest_api::{
+    ErrorResponse, HandlerFunction, Method, ProtocolVersionRangeGuard, Resource,
+};
+
+#[cfg(feature = "biome-profile")]
+use crate::biome::profile::store::{UserProfileStore, UserProfileStoreError};
+
+#[cfg(feature = "authorization")]
+use crate::biome::rest_api::BIOME_USER_READ_PERMISSION;
+
+/// Defines a REST endpoint to list profiles from the database
+pub fn make_profile_list_route(profile_store: Arc<dyn UserProfileStore>) -> Resource {
+    let resource =
+        Resource::build("/biome/profile/{id}").add_request_guard(ProtocolVersionRangeGuard::new(
+            protocol::BIOME_PROFILE_PROTOCOL_MIN,
+            protocol::BIOME_PROTOCOL_VERSION,
+        ));
+    #[cfg(feature = "authorization")]
+    {
+        resource.add_method(Method::Get, BIOME_USER_READ_PERMISSION, move |_, _| {
+            let profile_store = profile_store.clone();
+            Box::new(match profile_store.list_profiles() {
+                Ok(profiles) => HttpResponse::Ok().json(profiles).into_future(),
+                Err(err) => {
+                    debug!("Failed to get profiles from the database {}", err);
+                    HttpResponse::InternalServerError()
+                        .json(ErrorResponse::internal_error())
+                        .into_future()
+                }
+            })
+        })
+    }
+    #[cfg(not(feature = "authorization"))]
+    {
+        resource.add_method(Method::Get, move |_, _| {
+            let profile_store = profile_store.clone();
+            Box::new(match profile_store.list_profiles() {
+                Ok(profiles) => HttpResponse::Ok().json(profiles).into_future(),
+                Err(err) => {
+                    debug!("Failed to get profiles from the database {}", err);
+                    HttpResponse::InternalServerError()
+                        .json(ErrorResponse::internal_error())
+                        .into_future()
+                }
+            })
+        })
+    }
+}
+
+pub fn make_profile_routes(profile_store: Arc<dyn UserProfileStore>) -> Resource {
+    let resource =
+        Resource::build("/biome/profile/{id}").add_request_guard(ProtocolVersionRangeGuard::new(
```

Comment:
Looks like this path is incorrect (shouldn't have `/{id}`). Two other thoughts on these endpoints:
* I think they should be `/biome/profiles` and `/biome/profiles/{id}`; "profiles" plural because you're listing all profiles or getting one of the profiles in the system (this is a common naming convention for REST APIs).
* We probably need a `/biome/profile` endpoint that uses the authenticated user's identity to get their own profile information. I assume this is how UI will be getting information. Most users won't have the `BIOME_USER_READ_PERMISSION`, but should still be able to access their own profile info. This is how the biome key management endpoints work, for reference.

Reply:
Changed the existing endpoints to `/biome/profiles` and `/biome/profiles/{id}` and added `biome/profile` for retrieving the authenticated user's profile